synapse/util/stringutils.py

```diff
@@ -148,6 +148,11 @@ def valid_id_server_location(id_server: str) -> bool:
         True if valid, False otherwise.
     """
 
+    # A previous call to rewrite_id_server_url might have added a protocol scheme to the
+    # server's address - we don't care about that so remove it just for these checks.
+    if id_server.startswith("https://"):
+        id_server = id_server[8:]
```

Comment:
What about `http://`?

Also, is it possible to the cases where this method may be called with a protocol scheme, or is this really the only approach we have?

Reply:
Actually I've realised that in another place this function is called _before_ `rewrite_id_server_url`, so I'm just going to move it so it's always the case, that way we don't have to care about the protocol.